contentful/export-processor/content-types/section.js

```diff
@@ -152,6 +152,133 @@ export class Section {
+        const allAnswerPaths = [];
+
+        // Log ids of all answers in section
+        const allAnswers = this.questions.flatMap(question => question.answers.map(answer => answer.id));
+
+        // Log ids of answers used in minimum paths for recommendations
+        const allRecommendationPaths = this.minimumPathsForRecommendations;
+        const answersUsed = Object.values(allRecommendationPaths).flat().reduce((previous, current) => {
+            if (previous.indexOf(current.answer.id) === -1) {
+                previous.push(current.answer.id);
+            }
+            return previous;
+        }, []);
+
+        // Create paths until all answers are used
+        while (answersUsed.length !== allAnswers.length) {
+            let newPath = [];
+            const startIndex = this.questions.findLastIndex(question => question.answers.some(answer => !answersUsed.includes(answer.id)));
+
+            if (startIndex === -1) {
+                console.log("All answers have been used.");
+                break;
+            }
+
+            // Find the latest question in the sequence that has unused answers and assign an unused answer
+            const lastQuestionWithUnusedAnswers = this.questions[startIndex];
+            const lastUnusedAnswers = lastQuestionWithUnusedAnswers.answers.filter(answer => !answersUsed.includes(answer.id));
+            let latestAnswer = this.assignBestAnswer(lastUnusedAnswers, startIndex);
+            let nextNeeded = lastQuestionWithUnusedAnswers.id;
+
+            newPath.unshift({ question: lastQuestionWithUnusedAnswers, answer: latestAnswer });
+
+            // Loop backwards through questions leading up to the last with unused answers, adding answers that lead to needed questions
+            for (let i = startIndex - 1; i >= 0; i--) {
+                const currentQuestion = this.questions[i];
+                let currentAnswer;
+                const unusedAnswers = currentQuestion.answers.filter(answer => !answersUsed.includes(answer.id));
+
+                // Find answers that lead to the next question in the sequence, prioritising unused answers and skipping any that jump ahead
+                if (unusedAnswers.some(answer => answer.nextQuestion?.sys.id === nextNeeded)) {
+                    currentAnswer = unusedAnswers.find(answer => answer.nextQuestion.sys.id === nextNeeded);
+                } else if (currentQuestion.answers.some(answer => answer.nextQuestion.sys.id === nextNeeded)) {
+                    currentAnswer = currentQuestion.answers.find(answer => answer.nextQuestion.sys.id === nextNeeded);
+                } else {
+                    console.log(`No valid answer found for question ${currentQuestion.id}. Skipping.`);
+                }
+
+                if (currentAnswer) {
+                    nextNeeded = currentQuestion.id;
+                    newPath.unshift({ question: currentQuestion, answer: currentAnswer });
+                }
+            }
```

Comment:
I think this is possible to simplify by looking at all answers rather than only those of currentQuestion[i], then you wouldn't need to skip over various values of i, and could loop until the currentQuestion is the first one, so it reduces the 3 check cases to 2 and removes the need to check if currentAnswer is defined

This works though, so optional, feel free to resolve/defer

Reply:
Going to put a pin in this one for a future ticket.